File: src/main/java/com/eager/questioncloud/domain/workspace/implement/WorkSpaceQuestionUpdater.java
package com.eager.questioncloud.domain.workspace.implement;

import com.eager.questioncloud.domain.question.vo.QuestionContent;
import com.eager.questioncloud.domain.question.model.Question;
import com.eager.questioncloud.domain.question.repository.QuestionRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class WorkSpaceQuestionUpdater {
    private final QuestionRepository questionRepository;

    public void modify(Long creatorId, Long questionId, QuestionContent questionContent) {
        Question question = questionRepository.getForModifyAndDelete(questionId, creatorId);
        question.modify(questionContent);
        questionRepository.save(question);
    }
}

File: qc-core/qc-creator/qc-creator-api/src/main/kotlin/com/eager/questioncloud/workspace/implement/WorkspaceQuestionUpdater.kt
package com.eager.questioncloud.workspace.implement

import com.eager.questioncloud.common.exception.CoreException
import com.eager.questioncloud.common.exception.Error
import com.eager.questioncloud.creator.repository.CreatorRepository
import com.eager.questioncloud.question.api.internal.ModifyQuestionAPIRequest
import com.eager.questioncloud.question.api.internal.QuestionCommandAPI
import com.eager.questioncloud.workspace.command.ModifyQuestionCommand
import org.springframework.stereotype.Component

@Component
class WorkspaceQuestionUpdater(
    private val creatorRepository: CreatorRepository,
    private val questionCommandAPI: QuestionCommandAPI,
) {
    fun modifyQuestion(command: ModifyQuestionCommand) {
        val creator = creatorRepository.findByUserId(command.userId) ?: throw CoreException(Error.NOT_FOUND)
        questionCommandAPI.modify(
            creator.id,
            ModifyQuestionAPIRequest(
                command.questionCategoryId,
                command.subject,
                command.title,
                command.description,
                command.thumbnail,
                command.fileUrl,
                command.explanationUrl,
                command.questionLevel,
                command.price
            )
        )
    }
}